src/datastore/sync_methods/inject.js

```diff
@@ -313,12 +313,10 @@ module.exports = function inject (resourceName, attrs, options) {
     definition.emit('DS.beforeInject', definition, attrs)
   }
 
-  // start the recursive injection of data
+  // start the recursive injection of data; this will also call
+  // `definition.handleChange` for each item
   injected = _inject.call(_this, definition, resource, attrs, options)
 
-  // collection was modified
```

Comment:
These lines should not have been removed. Without these lines, `DS.change` will never be emitted on the Resource.